command-parser.js

```diff
@@ -634,7 +635,8 @@ let parse = exports.parse = function (message, room, user, connection, levelsDee
 	});
 
 	if (commandHandler) {
-		return context.run(commandHandler);
+		message = context.run(commandHandler);
+		if (!message || typeof message.then === 'function') return message;
```

Comment:
It's weird how much easier this code is to understand when I have the full code in front of me and am trying to fix it myself, instead of just a diff.

Anyway, this line is redundant, line 676 (old) aka 678 (new) are already doing this job.